src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
```

Comment:
on the implementation details: `step` in this component would become numerical
```suggestion
  const [currentStep, setStep] = useState<number>(0)
  const goToNextStep = useCallback(() => setStep(step => step + 1), [])
```

Reply:
One thing I've missed to add right from the beginning is the necessity to have a mapping for sequence step - component pairs:

```ts
export const stepComponentMapping = {
  email: EmailStep,
  age: AgeStep,
  summary: SummaryStep,
}
```